tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptioncomponents/SubscriptionComponentsUpdatePrepaidAllocationExpirationDateTest.java

```diff
@@ -0,0 +1,195 @@
+import com.maxio.advancedbilling.utils.TestSetup;
+import com.maxio.advancedbilling.utils.TestTeardown;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.stream.Stream;
+
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertNotFound;
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertUnauthorized;
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertUnprocessableEntity;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SubscriptionComponentsUpdatePrepaidAllocationExpirationDateTest {
+    private static final TestSetup TEST_SETUP = new TestSetup();
+    private static final SubscriptionComponentsController SUBSCRIPTION_COMPONENTS_CONTROLLER =
+            TestClient.createClient().getSubscriptionComponentsController();
+
+    private static Component prepaidComponent;
+    private static Allocation allocation;
+    private static Customer customer;
+    private static Subscription subscription;
+
+    @BeforeAll
+    static void setup() throws IOException, ApiException {
+        ProductFamily productFamily = TEST_SETUP.createProductFamily();
+        customer = TEST_SETUP.createCustomer();
+        prepaidComponent = TEST_SETUP.createPrepaidComponent(productFamily, customizer -> customizer
+                .rolloverPrepaidRemainder(true)
+                .expirationIntervalUnit(IntervalUnit.DAY)
+                .expirationInterval(10.0)
+        );
+        subscription = TEST_SETUP.createSubscription(customer, TEST_SETUP.createProduct(productFamily));
+        allocation = SUBSCRIPTION_COMPONENTS_CONTROLLER.allocateComponent(
+                subscription.getId(),
+                prepaidComponent.getId(),
+                new CreateAllocationRequest(new CreateAllocation.Builder()
+                        .quantity(10.0)
+                        .memo("Recoding component allocation")
+                        .build()
+                )
+        ).getAllocation();
+    }
+
+    @AfterAll
+    static void teardown() throws IOException, ApiException {
+        new TestTeardown().deleteCustomer(customer);
+    }
+
+    @Test
+    void shouldUpdatePrepaidUsageAllocationExpirationDate() throws IOException, ApiException {
+        // given
+        ZonedDateTime updatedExpiresAt = ZonedDateTime.now().plusMonths(2);
+
+        // when
+        SUBSCRIPTION_COMPONENTS_CONTROLLER.updatePrepaidUsageAllocationExpirationDate(
+                subscription.getId(),
+                prepaidComponent.getId(),
+                allocation.getAllocationId(),
+                // changing from 10 to 5 days
```

Comment:
why 10 to 5 days, isn't it +2 months?

Reply:
In line 55, I've set the `.expirationInterval(10.0)` and decided here to decrease it to `5 days` so, I don't get a question because it's a matter of client what value is supposed to be configured.

Reply:
`updatedExpiresAt = ZonedDateTime.now().plusMonths(2);` is used here, so I thought you're adding about 60 days, and I don't understand from where `5` comes from

Reply:
you got `new UpdateAllocationExpirationDate(new AllocationExpirationDate(ZonedDateTime.now().plusDays(5)))` in another test, maybe you mean this one?

Reply:
Ahh, right 👍 Sorry for my oversight :D It's just a leftover from some old version.